src/ucs/datastruct/conn_match.h

```diff
@@ -43,81 +37,86 @@ typedef enum ucs_conn_match_queue_type {
      * connected to remote peer, but not provided to user yet */
-    UCS_CONN_MATCH_QUEUE_UNEXP
+    UCS_CONN_MATCH_QUEUE_UNEXP,
+    /* Number of queues that are used by connection matching */
+    UCS_CONN_MATCH_QUEUE_LAST
 } ucs_conn_match_queue_type_t;
 
 
 /**
  * Structure to embed in a connection entry to support matching with remote
  * peer's connections
  */
-typedef struct ucs_conn_match {
+typedef struct ucs_conn_match_elem {
     ucs_hlist_link_t          list;          /* List entry into endpoint
                                                 matching structure */
-} ucs_conn_match_t;
+} ucs_conn_match_elem_t;
 
 
 /**
- * Connection entry - allows *ordered* matching of connections between a pair
- * of connected peers. The expected/unexpected lists are *not* circular
+ * Connection peer entry - allows *ordered* matching of connections between
+ * a pair of connected peers. The expected/unexpected lists are *not* circular
  */
-typedef struct ucs_conn {
-    ucs_hlist_head_t         conn_q[2];       /* Connection queues */
-    ucs_conn_sn_t            next_conn_sn;    /* Sequence number of matching
-                                                 connections, since wireup messages
-                                                 used for connection establishment
-                                                 procedure which were sent on different
-                                                 connections could be provided
-                                                 out-of-order */
-} ucs_conn_t;
+typedef struct ucs_conn_match_peer {
+    ucs_hlist_head_t         conn_q[UCS_CONN_MATCH_QUEUE_LAST]; /* Connection queues */
+    ucs_conn_sn_t            next_conn_sn;                      /* Sequence number of matching
+                                                                   connections, since wireup messages
+                                                                   used for connection establishment
+                                                                   procedure which were sent on different
+                                                                   connections could be provided
+                                                                   out-of-order */
+} ucs_conn_match_peer_t;
 
 
-KHASH_TYPE(ucs_conn_match, ucs_conn_match_addr_t, ucs_conn_t)
+KHASH_TYPE(ucs_conn_match, ucs_conn_match_address_t, ucs_conn_match_peer_t)
 
 
 /**
  * Function to get the address of the connection between the peers.
  *
- * @param [in]  conn_match    Pointer to the connection matching.
- * @param [out] addr_p        Pointer to the address structure that
+ * @param [in]  elem          Pointer to the connection matching element.
+ * @param [out] address_p     Pointer to the address structure that
  *                            is filled by the address between the peers.
+ *                            User is responsible to allocate the address
```

Comment:
so user allocates, but the conn_match implementation releases?
https://github.com/openucx/ucx/pull/5330/commits/55eefdea8ff4e7c6e5281097dbbc8e525da0882f#diff-43e8751be148f6a14e1c327ba4ddc788R74
seems problematic

Reply:
no need anymore